presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/MergeLimitWithDistinct.java

```diff
@@ -65,4 +72,11 @@ public Result apply(LimitNode parent, Captures captures, Context context)
                         child.getGroupingKeys(),
                         child.getHashVariable()));
     }
+
+    @Override
+    public boolean isEnabled(Session session)
+    {
+        allowedLimitThreshold = SystemSessionProperties.getDistinctLimitOperatorThreshold(session);
```

Comment:
This doesn't look clean. I say do it in the constructor in PlanOptimizers or something.